### PR TITLE
[WIP] Fix 0 destination node error in HGT encoder

### PR DESCRIPTION
**Pull request by zhjwy9343**
*Issue #, if available:*

*Description of changes:*
This PR fix the issue of mismatch between number of nodes and number of features when the number of destination nodes is 0.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/model/hgt_encoder.py [hunk 1]
```diff
@@ -320,7 +320,8 @@ def forward(self, g, h):
                 else:
                     # Handle zero number of dst nodes, which is an extreme case
                     if g.dstnodes[k].data.get('t') is not None:
-                        trans_out = self.a_linears[k](h[k])
+                        dst_h_k = g.dstnodes[k].data.get('t').view(-1, self.out_dim)
```
Comment (classicsong) [inline python/graphstorm/model/hgt_encoder.py:323]: If h[k] does not exist, can we just return a tensor with shape `(0, self.out_dim)`? There is no need to call `self.a_linears[k](h)`.

**Comment by jalencato** [comment]
Do we have any profiling result with this new HGT?

File: tests/unit-tests/test_nn_model.py [hunk 2]
```diff
@@ -103,6 +103,36 @@ def create_dummy_no_indegree_test_graph(dim=16):
 
     return block, inputs, list(num_src_nodes.keys()), list(edges.keys())
 
+
+def create_dummy_zero_dest_test_graph(dim):
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply tests/unit-tests/test_nn_model.py:107]: Yes, in DGL 2.0+, there are such edges in MFCs.

**Comment by zhjwy9343** [comment]
This change does not increase computations. So the run time profile should be equal or smaller than the previous version. And because this PR fix a corner case where no real embeddings will be used for decoders, the model performance profile should be equal to the previous version.

File: tests/unit-tests/test_nn_model.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (classicsong) [reply tests/unit-tests/test_nn_model.py:107]: I mean in this test case, you only cover the case when dst is 0. What about src is also 0?
